src/main/java/it/pagopa/ecommerce/commons/repositories/PaymentRequestInfo.java

```diff
@@ -19,25 +19,19 @@
  * @param description    Description of the payment notice
  * @param amount         Amount on the payment notice
  * @param dueDate        Payment's due date
- * @param isNM3          If true, this payment notice must go through the new
- *                       payment flow (Nuovo Modello 3) instead of the legacy
- *                       one.
- * @param paymentToken   Payment token associated to this payment request. May
- *                       be null iff {@code
- *                       isNM3} is {@code false}
+ * @param paymentToken   Payment token associated to this payment request
  * @param idempotencyKey Idempotency key associated to the payment request
  */
 @RedisHash(value = "keys", timeToLive = 10 * 60)
 public record PaymentRequestInfo(
         @NonNull @Id RptId id,
-        @NonNull String paFiscalCode,
+        @Nullable String paFiscalCode,
         @Nullable String paName,
         @Nullable String description,
-        @NonNull Integer amount,
+        @Nullable Integer amount,
         @Nullable String dueDate,
-        @NonNull Boolean isNM3,
         @Nullable String paymentToken,
-        @NonNull IdempotencyKey idempotencyKey
+        @Nullable IdempotencyKey idempotencyKey
```

Comment:
Shouldn't this be always populated?

Reply:
no, in the case of `GET payment-requests` the idempotency key is not generated (it is generated with `POST transactions)`